File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/entity/PedidoProduto.java
package br.edu.utfpr.serverpedidosjava.entity;

import java.math.BigDecimal;

import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;

@Entity
public class PedidoProduto {
    @EmbeddedId
    private PedidoProdutoId id;
    private BigDecimal valorUnitario = BigDecimal.ZERO;
    private BigDecimal quantidade = BigDecimal.ZERO;

    public PedidoProdutoId getId() {
        return id;
    }

    public void setId(PedidoProdutoId id) {
        this.id = id;
    }

    public BigDecimal getValorUnitario() {
        return valorUnitario;
    }

    public void setValorUnitario(BigDecimal valorUnitario) {
        this.valorUnitario = valorUnitario;
    }

    public BigDecimal getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(BigDecimal quantidade) {
        this.quantidade = quantidade;
    }
}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/entity/PedidoProduto.kt
package br.edu.utfpr.serverpedidos.entity

import jakarta.persistence.EmbeddedId
import jakarta.persistence.Entity
import java.math.BigDecimal

@Entity
class PedidoProduto(
    @EmbeddedId
    var id: PedidoProdutoId,
    var valorUnitario: BigDecimal = BigDecimal.ZERO,
    var quantidade: BigDecimal = BigDecimal.ZERO
)
